src/analysis/retail/evoker/preservation/modules/talents/ConsumeFlame.tsx

```diff
@@ -0,0 +1,133 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import { SpellLink } from 'interface';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import { BoxRowEntry, PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import Combatants from 'parser/shared/modules/Combatants';
+import { getDreamBreathCast } from '../../normalizers/EventLinking/helpers';
+
+interface CastInfo {
+  timestamp: number;
+  coyActive: boolean;
+  temporalCompressionStacked: boolean;
+  dreamBreathOnTarget: boolean;
+}
+
+class ConsumeFlame extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  protected combatants!: Combatants;
+
+  casts: CastInfo[] = [];
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell([TALENTS_EVOKER.ENGULF_TALENT]),
+      this.onCast,
+    );
+  }
+
+  onCast(event: CastEvent) {
+    const target = this.combatants.getEntity(event);
+    let dreamBreathOnTarget = true;
+
+    if (!target?.hasBuff(SPELLS.DREAM_BREATH.id) && !target?.hasBuff(SPELLS.DREAM_BREATH_ECHO.id)) {
+      dreamBreathOnTarget = false;
+    }
+
+    const applyBuffEvent = target?.getBuff(SPELLS.DREAM_BREATH.id);
+    console.log(applyBuffEvent?._linkedEvents);
+
+    let cast;
+
+    if (applyBuffEvent) {
+      cast = getDreamBreathCast(applyBuffEvent, false);
+    }
+
+    const temporalCompressionStacks = this.selectedCombatant?.getBuffStacks(
+      SPELLS.TEMPORAL_COMPRESSION_BUFF,
+      cast?.timestamp,
+    );
+    const coyActive = this.selectedCombatant.hasBuff(
+      SPELLS.CALL_OF_YSERA_BUFF.id,
+      applyBuffEvent?.timestamp,
+      150,
+    );
+
+    this.casts.push({
+      timestamp: event.timestamp,
+      dreamBreathOnTarget: dreamBreathOnTarget,
+      temporalCompressionStacked: temporalCompressionStacks === 4,
```

Comment:
I think that storing the number of stacks and then making 3 stacks be marked as `QualitativePerformance.Ok` would be a bit more lenient.

Also, this should only be checked if the player has 4pc.